pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -524,8 +530,8 @@ func loadBalancerServiceTagsModified(current, expected *corev1.Service) (bool, *
 // return value is nil.  Otherwise, if something or someone else has modified
 // the service, then the return value is a non-nil error indicating that the
 // modification must be reverted before upgrading is allowed.
-func loadBalancerServiceIsUpgradeable(ic *operatorv1.IngressController, deploymentRef metav1.OwnerReference, current *corev1.Service, platform *configv1.PlatformStatus) error {
-	want, desired, err := desiredLoadBalancerService(ic, deploymentRef, platform)
+func loadBalancerServiceIsUpgradeable(ic *operatorv1.IngressController, deploymentRef metav1.OwnerReference, current *corev1.Service, platform *configv1.PlatformStatus, networkConfig *configv1.Network) error {
```

Comment:
@danwinship or @squeed : I have always been confused as to when to use the operv1 versus configv1 api for things. Could you please configm if using configv1 ^ is ok here? Seems like for config we support sdn and ovn: https://github.com/openshift/api/blob/d5252bac47154049f80d49eff7fec5bb642be9af/config/v1/types_network.go#L145 and for operator we support the other ones like kuryr etc..sorry for the dumb question, I probably should be knowing these things by now.

Reply:
It is confusing that `operator/v1/types_network.go` defines these named constants, but `config/v1/types_network.go` just uses a string type.

Reply:
@tssurya good question! In this case, it doesn't really matter, because configv1 NetworkType is inferred from configv1. Since configv1 is guaranteed to be set from cluster creation, it is slightly better to use. But it really, really, really doesn't matter.

Reply:
> It is confusing that `operator/v1/types_network.go` defines these named constants [but configv1 doesn't]

That's because you can run a cluster with a third-party network operator - so we need to support arbitrary network types in configv1. But operv1 defines what the openshift cluster network operator supports.

Reply:
>  because configv1 NetworkType is inferred from configv1. 

you meant from operv1?

> Since configv1 is guaranteed to be set from cluster creation, it is slightly better to use. But it really, really, really doesn't matter. 

ack thanks casey! makes sense. @Miciah so I think we are good here.